crates/rome_js_analyze/src/analyzers/nursery/no_misused_new.rs

```diff
@@ -0,0 +1,232 @@
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        match node {
+            DeclarationQuery::TsInterfaceDeclaration(decl) => {
+                let interface_ident = decl.id().ok()?.name_token().ok()?;
+                for member in decl.members() {
+                    match member {
+                        AnyTsTypeMember::TsConstructSignatureTypeMember(construct) => {
+                            if construct.new_token().ok().is_none() {
+                                continue;
+                            }
+                            let any_ts_type = construct.type_annotation()?.ty().ok()?;
+                            let reference_type = any_ts_type.as_ts_reference_type()?;
+                            let return_type_ident = extract_return_type_ident(&reference_type)?;
+                            if interface_ident.text_trimmed() == return_type_ident.text_trimmed() {
+                                return Some(RuleState::InterfaceMisusedNew(construct.range()));
+                            }
+                        }
+                        AnyTsTypeMember::TsMethodSignatureTypeMember(method) => {
+                            let method_name = method.name().ok()?;
+                            let method_name = method_name.as_js_literal_member_name()?.syntax();
+                            if method_name.text_trimmed() == "constructor" {
+                                return Some(RuleState::InterfaceMisusedConstructor(
+                                    method.range(),
+                                ));
+                            }
+                        }
+                        _ => continue,
+                    };
+                }
+                None
+            }
+            DeclarationQuery::TsTypeAliasDeclaration(decl) => {
+                let any_ts_type = decl.ty().ok()?;
+                let object = any_ts_type.as_ts_object_type()?;
+                let method = object
+                    .members()
+                    .into_iter()
+                    .filter_map(|member| member.as_ts_method_signature_type_member().cloned())
+                    .next()?;
+                let method_name = method.name().ok()?;
+                let method_name = method_name.as_js_literal_member_name()?.syntax();
+                if method_name.text_trimmed() == "constructor" {
+                    return Some(RuleState::TypeAliasMisusedConstructor(method.range()));
+                }
+                None
+            }
+            DeclarationQuery::JsClassDeclaration(decl) => {
+                check_new_method_in_class(decl).map(|range| RuleState::ClassMisusedNew(range))
+            }
+            DeclarationQuery::TsDeclareStatement(decl) => {
+                let decl = decl.declaration().ok()?;
+                let decl = decl.as_js_class_declaration()?;
```

Comment:
This class declaration should already be handled in `DeclarationQuery::JsClassDeclaration`? Am I missing something?

Reply:
That's not handled by `JsClassDeclaration` but I don't know why, need to investigate it.